### DISCUSSION TITLE
hybrid-bonded network type is not listed in the documentation

### DISCUSSION THREAD

**Discussion by ocobles** [equinix/terraform-provider-metal#50] [Q&A] [answered]
Hi,

(I'm not sure if this is the place to report documentation related suggestions / issues)

I see "hybrid-bonded" is available in 1.1.0 code but it is not listed in the resource documentation as valid type. Shouldn't it appear there too?

https://github.com/equinix/terraform-provider-metal/blob/v1.1.0/docs/resources/device_network_type.md

Thanks

**Comment by t0mk** [comment]
hi @ocobleseqx thanks for bringin this up. The provider doesn't support the `hybrid-bonded` mode yet. There's some ongoing effort:

* https://github.com/equinix/terraform-provider-metal/issues/12 (maybe this is the place to continue discussion)
* https://github.com/packethost/packngo/pull/239

**Comment by displague** [comment, marked as answer]
@ocobleseqx as @t0mk there are some other ideas swirling around about how best to present `network_type` in this provider. The provider is following the pattern established by the Equinix Metal web console interface by offering a simple "Network Type" toggle.

At the API level, which is the level Terraform should strive to match, there is no device wide network type. This model is too simple to express the permutations of port configurations available on 4 port devices with a mix of Layer2/Layer3/Hybrid, bonded/individual, and VLAN attached / publicly addressable settings.  Per the issues linked above, in the not too distant future, we may adopt a pattern that brings Port configuration to the surface.

You are right that `hybrid-bonded` should be mentioned in the docs. The implementation in 1.1.0 could be described as one that tolerates `hybrid-bonded` but does not need to acknowledge that it is a new setting.

This is because, from Terraform and the API's perspective, `hybrid-bonded` is effectively the same as `layer3` (the default network_type) with the added ability to add and remove [metal_port_vlan_attachment](https://registry.terraform.io/providers/equinix/metal/latest/docs/resources/port_vlan_attachment) resources without an explicit `metal_device_network_type`.

We should also provide an example in the documentation that shows how to use `hybrid-bonded` explicitly. We should also provide an example of implicit use with the `layer3` default.
